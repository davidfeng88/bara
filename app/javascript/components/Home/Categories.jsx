import React from 'react';

const Categories = () => {
  const categoriesTitle = <h2>Browse Businesses by Category</h2>;
  return (
    <div className="center home-categories">
      {categoriesTitle}
      <CategoryCards />
    </div>
  );
};

const CategoryCards = () => (
  <div className="category-cards">
    <RestaurantCategory />
    <NightlifeCategory />
  </div>
);

const RestaurantCategory = () => (
  <a
    className="category-card"
    href="#/businesses/?name=&location=New%20York"
  >
    <img alt="" src={window.staticImages.restaurants} />
    <p>Restaurants</p>
  </a>
);

const NightlifeCategory = () => (
  <a
    className="category-card"
    href="#/businesses/?name=&location=New%20York&tag=nightlife"
  >
    <img alt="" src={window.staticImages.nightlife} />
    <p>Nightlife</p>
  </a>
);

export default Categories;
