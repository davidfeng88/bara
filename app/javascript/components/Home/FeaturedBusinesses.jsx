import React from 'react';
import Rating from 'react-rating';
import PropTypes from 'prop-types';
import {
  nReviewsString,
  priceToDollarSign,
  tagContent,
  LoadingSpinner,
} from '../../util/BusinessInfoUtil';

const FeaturedBusinesses = ({
  featuredBusinesses,
  isLoading,
}) => {
  if (isLoading) {
    return <LoadingSpinner />;
  }
  const title = <h2>Featured Businesses</h2>;
  return (
    <div className="center home-businesses">
      {title}
      <FeaturedBusinessesList featuredBusinesses={featuredBusinesses} />
    </div>
  );
};

export default FeaturedBusinesses;

FeaturedBusinesses.propTypes = {
  featuredBusinesses: PropTypes.arrayOf(PropTypes.object).isRequired,
  isLoading: PropTypes.bool.isRequired,
};

const FeaturedBusinessesList = ({
  featuredBusinesses,
}) => (
  <div className="home-businesses-list">
    {featuredBusinesses.map(business =>
      (<FeaturedBusinessListItem
        key={business.id}
        business={business}
      />))
    }
  </div>
);

FeaturedBusinessesList.propTypes = {
  featuredBusinesses: PropTypes.arrayOf(PropTypes.object).isRequired,
};

const FeaturedBusinessListItem = ({
  business,
}) => {
  const {
    id,
    image,
    name,
    averageRating,
    numberOfReviews,
    price,
    tags,
  } = business;
  return (
    <div className="home-business-item">
      <Pic
        id={id}
        image={image}
      />
      <FeatureTextInfo
        id={id}
        name={name}
        averageRating={averageRating}
        numberOfReviews={numberOfReviews}
        price={price}
        tags={tags}
      />
    </div>
  );
};

FeaturedBusinessListItem.propTypes = {
  business: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    averageRating: PropTypes.string,
    numberOfReviews: PropTypes.number,
    price: PropTypes.number,
  }).isRequired,
};

const Pic = ({
  id,
  image,
}) => (
  <a href={`#/businesses/${id}`}>
    <img alt="" src={image} />
  </a>
);

Pic.propTypes = {
  id: PropTypes.number.isRequired,
  image: PropTypes.string.isRequired,
};

const FeatureTextInfo = ({
  id,
  name,
  averageRating,
  numberOfReviews,
  price,
  tags,
}) => (
  <div className="card-content">
    <BusinessNameLink
      id={id}
      name={name}
    />
    <RatingInfo
      averageRating={averageRating}
      numberOfReviews={numberOfReviews}
    />
    {priceToDollarSign[price]}
    {tagContent(tags)}
  </div>
);

FeatureTextInfo.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  averageRating: PropTypes.string.isRequired,
  numberOfReviews: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
  tags: PropTypes.arrayOf(PropTypes.object),
};

FeatureTextInfo.defaultProps = {
  tags: null,
};

const BusinessNameLink = ({
  id,
  name,
}) => (
  <a href={`#/businesses/${id}`}>
    <strong>{name}</strong>
  </a>
);

BusinessNameLink.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
};

const RatingInfo = ({
  averageRating,
  numberOfReviews,
}) => (
  <div>
    <Rating
      className="rating"
      emptySymbol="fa fa-star-o fa-lg"
      fullSymbol="fa fa-star fa-lg"
      initialRating={parseFloat(averageRating)}
      readonly
    />
    {nReviewsString(numberOfReviews)}
  </div>
);

RatingInfo.propTypes = {
  averageRating: PropTypes.string.isRequired,
  numberOfReviews: PropTypes.number.isRequired,
};
