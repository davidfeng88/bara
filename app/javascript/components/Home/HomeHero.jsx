import React from 'react';
import PropTypes from 'prop-types';
import HomeHeaderContainer from './HomeHeader/HomeHeaderContainer';
import SearchBar from '../Header/SearchBar';
import HomeLinks from './HomeLinks';

const HomeLogo = ({
  handleHomeLogoClick,
}) => (
  <div
    className="logo"
    onClick={handleHomeLogoClick}
    onKeyPress={handleHomeLogoClick}
    role="button"
    tabIndex={0}
  >
    <img alt="home logo" src={window.staticImages.homeLogo} />
  </div>
);

HomeLogo.propTypes = {
  handleHomeLogoClick: PropTypes.func.isRequired,
};

const HomeSearch = () => (
  <div className="home-search">
    <SearchBar />
  </div>
);

const HomeHero = ({
  handleHomeLogoClick,
  hasDefaultBackground,
}) => {
  const homeHeroBackgroundCSSClassName =
    hasDefaultBackground ?
      'home-hero-bg-1' :
      'home-hero-bg-2';
  return (
    <div className={homeHeroBackgroundCSSClassName}>
      <HomeHeaderContainer />
      <HomeLogo handleHomeLogoClick={handleHomeLogoClick} />
      <HomeSearch />
      <HomeLinks />
    </div>
  );
};

HomeHero.propTypes = {
  handleHomeLogoClick: PropTypes.func.isRequired,
  hasDefaultBackground: PropTypes.bool.isRequired,
};

export default HomeHero;
