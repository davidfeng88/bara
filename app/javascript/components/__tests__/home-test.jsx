// run test: npm test
// update snapshots: npm test -- -u

import React from 'react';
import renderer from 'react-test-renderer';
import HomeHero from '../home/HomeHero';
import FeaturedBusinesses from '../home/FeaturedBusinesses';
import Categories from '../home/Categories';

jest.mock('react-rating');
jest.mock('../header/SearchBar', () => ('SearchBar Mock'));
jest.mock('../home/home_header/HomeHeaderContainer', () => ('HomeHeaderContainer Mock'));

window.staticImages = {};

const businesses = [{
  id: 464,
  author_id: 530,
  name: 'Buddakan',
  address: '75 9th Ave',
  city: 'New York',
  state: 'NY',
  zipcode: 10011,
  lat: 40.742287,
  lng: -74.004819,
  price: 3,
  url: 'buddakannyc.com',
  phone: '(212) 989-6699',
  averageRating: '3.1428571428571429',
  tags: [{
    id: 17,
    label: 'Chinese',
    created_at: '2017-09-02T21:10:59.608Z',
    updated_at: '2017-09-02T21:10:59.608Z',
  },
  {
    id: 22,
    label: 'Asian Fusion',
    created_at: '2017-09-02T21:10:59.629Z',
    updated_at: '2017-09-02T21:10:59.629Z',
  },
  ],
  imageUrl: 'http://s3.amazonaws.com/bara-dev/businesses/images/000/000/464/original/Chinese.jpg?1504386660',
  numberOfReviews: 14,
},
{
  id: 460,
  author_id: 542,
  name: 'Dovetail',
  address: '103 W 77th St',
  city: 'New York',
  state: 'NY',
  zipcode: 10024,
  lat: 40.780927,
  lng: -73.976625,
  price: 4,
  url: 'dovetailnyc.com',
  phone: '(212) 362-3800',
  averageRating: '4.5714285714285714',
  tags: [{
    id: 20,
    label: 'American',
    created_at: '2017-09-02T21:10:59.621Z',
    updated_at: '2017-09-02T21:10:59.621Z',
  },
  {
    id: 21,
    label: 'French',
    created_at: '2017-09-02T21:10:59.625Z',
    updated_at: '2017-09-02T21:10:59.625Z',
  },
  ],
  imageUrl: 'http://s3.amazonaws.com/bara-dev/businesses/images/000/000/460/original/French.jpg?1504386660',
  numberOfReviews: 14,
},
{
  id: 462,
  author_id: 530,
  name: 'Burger & Lobster',
  address: '39 W 19th St',
  city: 'New York',
  state: 'NY',
  zipcode: 10011,
  lat: 40.740123,
  lng: -73.993345,
  price: 2,
  url: 'burgerandlobster.com/en',
  phone: '(646) 833-7532',
  averageRating: '4.2727272727272727',
  tags: [{
    id: 20,
    label: 'American',
    created_at: '2017-09-02T21:10:59.621Z',
    updated_at: '2017-09-02T21:10:59.621Z',
  },
  {
    id: 19,
    label: 'Seafood',
    created_at: '2017-09-02T21:10:59.616Z',
    updated_at: '2017-09-02T21:10:59.616Z',
  },
  ],
  imageUrl: 'http://s3.amazonaws.com/bara-dev/businesses/images/000/000/462/original/Seafood.jpg?1504386660',
  numberOfReviews: 11,
},
];

it('renders HomeHero correctly', () => {
  const tree = renderer
    .create(<HomeHero
      handleHomeLogoClick={() => {}}
      hasDefaultBackground
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders FeaturedBusinesses correctly', () => {
  const tree = renderer
    .create(<FeaturedBusinesses
      featuredBusinesses={businesses}
      isLoading={false}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Categories correctly', () => {
  const tree = renderer
    .create(<Categories />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
