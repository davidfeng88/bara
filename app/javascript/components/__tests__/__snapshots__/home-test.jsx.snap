// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Categories correctly 1`] = `
<div
  className="center home-categories"
>
  <h2>
    Browse Businesses by Category
  </h2>
  <div
    className="category-cards"
  >
    <a
      className="category-card"
      href="#/businesses/?name=&location=New%20York"
    >
      <img
        alt=""
        src={undefined}
      />
      <p>
        Restaurants
      </p>
    </a>
    <a
      className="category-card"
      href="#/businesses/?name=&location=New%20York&tag=nightlife"
    >
      <img
        alt=""
        src={undefined}
      />
      <p>
        Nightlife
      </p>
    </a>
  </div>
</div>
`;

exports[`renders FeaturedBusinesses correctly 1`] = `
<div
  className="center home-businesses"
>
  <h2>
    Featured Businesses
  </h2>
  <div
    className="home-businesses-list"
  >
    <div
      className="home-business-item"
    >
      <a
        href="#/businesses/464"
      >
        <img
          alt=""
          src="http://s3.amazonaws.com/bara-dev/businesses/images/000/000/464/original/Chinese.jpg?1504386660"
        />
      </a>
      <div
        className="card-content"
      >
        <a
          href="#/businesses/464"
        >
          <strong>
            Buddakan
          </strong>
        </a>
        <div>
          14 reviews
        </div>
        $$$
         • 
        <a
          href="/businesses/?tag=Chinese"
        >
          Chinese
        </a>
        , 
        <a
          href="/businesses/?tag=Asian%20Fusion"
        >
          Asian Fusion
        </a>
      </div>
    </div>
    <div
      className="home-business-item"
    >
      <a
        href="#/businesses/460"
      >
        <img
          alt=""
          src="http://s3.amazonaws.com/bara-dev/businesses/images/000/000/460/original/French.jpg?1504386660"
        />
      </a>
      <div
        className="card-content"
      >
        <a
          href="#/businesses/460"
        >
          <strong>
            Dovetail
          </strong>
        </a>
        <div>
          14 reviews
        </div>
        $$$$
         • 
        <a
          href="/businesses/?tag=American"
        >
          American
        </a>
        , 
        <a
          href="/businesses/?tag=French"
        >
          French
        </a>
      </div>
    </div>
    <div
      className="home-business-item"
    >
      <a
        href="#/businesses/462"
      >
        <img
          alt=""
          src="http://s3.amazonaws.com/bara-dev/businesses/images/000/000/462/original/Seafood.jpg?1504386660"
        />
      </a>
      <div
        className="card-content"
      >
        <a
          href="#/businesses/462"
        >
          <strong>
            Burger & Lobster
          </strong>
        </a>
        <div>
          11 reviews
        </div>
        $$
         • 
        <a
          href="/businesses/?tag=American"
        >
          American
        </a>
        , 
        <a
          href="/businesses/?tag=Seafood"
        >
          Seafood
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders HomeHero correctly 1`] = `
<div
  className="home-hero-bg-1"
>
  <HomeHeaderContainer Mock />
  <div
    className="logo"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  >
    <img
      alt="home logo"
      src={undefined}
    />
  </div>
  <div
    className="home-search"
  >
    <SearchBar Mock />
  </div>
  <div
    className="home-header-links"
  >
    <span
      className="home-header-link"
    >
      <i
        aria-hidden="true"
        className="fa fa-cutlery"
      />
      <a
        href="#/businesses/?name=&location=New%20York"
      >
        Restaurants
      </a>
    </span>
    <span
      className="home-header-link"
    >
      <i
        aria-hidden="true"
        className="fa fa-glass"
      />
      <a
        href="#/businesses/?name=&location=New%20York&tag=nightlife"
      >
        Nightlife
      </a>
    </span>
    <p>
      Search Example: 
      <a
        href="#/businesses/?name=burger&location=19th%20St"
      >
        find "burger" near "19th St"
      </a>
    </p>
    <p>
      Click on bara logo for new random featured businesses
    </p>
  </div>
</div>
`;
